src/components/Details/Details.js

```diff
@@ -0,0 +1,208 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import PropTypes from 'prop-types'
+import { Spring, animated } from 'react-spring'
+import { useInside } from 'use-inside'
+import { GU, springs, textStyle } from '../../style'
+import { useTheme } from '../../theme'
+import { IconDown } from '../../icons'
+import ButtonBase from '../ButtonBase/ButtonBase'
+
+// Interpolate the elevation of the toggle from which the drawer slides down.
+// In / out example: [0, 0.25, 0.5, 0.75, 1] => [0, 0.5, 1, 0.5, 0]
+function interpolateToggleElevation(value, fn) {
+  return value.interpolate(v => fn(1 - Math.abs(v * 2 - 1)))
+}
+
+function Details({ children, label, onToggle, opened: openedProp, ...props }) {
+  const theme = useTheme()
+  const [insideSidePanel] = useInside('SidePanel')
+
+  const contentRef = useRef(null)
+  const contentHeight = useRef(0)
+
+  // Details supports two modes: managed (internal state),
+  // or controlled (external state).
+  const [openedManaged, setOpenedManaged] = useState(false)
+  const opened = openedProp === undefined ? openedManaged : openedProp
+
+  const updateHeight = useCallback(() => {
+    if (contentRef.current) {
+      contentHeight.current = contentRef.current.clientHeight
+    }
+  }, [])
+
+  const handleContentRef = useCallback(
+    element => {
+      contentRef.current = element
+      updateHeight()
+    },
+    [updateHeight]
+  )
+
+  const handleToggle = useCallback(() => {
+    const newOpened = !opened
+
+    // Managed state
+    if (openedProp === undefined) {
+      setOpenedManaged(newOpened)
+    }
+
+    // Useful to notify even in managed mode
+    if (onToggle) {
+      onToggle(newOpened)
+    }
+  }, [onToggle, opened, openedProp])
+
+  // Animate after the initial render
+  const animate = useRef(false)
+  useEffect(() => {
+    animate.current = true
+  }, [])
+
+  // Use height: 'auto' when opened
+  const [forceHeight, setForceHeight] = useState(false)
+  const handleRest = useCallback(() => setForceHeight(!opened), [opened])
```

Comment:
```suggestion
  const handleRest = useCallback(() => setForceHeight(opened => !opened), [])
```

Would this also work?

Reply:
It would yes :+1:

Reply:
…no it doesn’t haha, we need to use the `opened` from the previous render cycle. I reverted => https://github.com/aragon/aragon-ui/pull/729/commits/97906a11086214c358b5ef3e1d00288056282a84

Reply:
Sounds good, might be useful to leave a comment!